### PR TITLE
Benchmarks for get_qsp_time_evolution_program

### PR DESCRIPTION
**Pull request by dexter2206**
## Description

This adds simple benchmarks for `get_qsp_time_evolution_program` (prevoiusly `get_qsp_program`). Benchmarks might take a while to run, and hence I've decided to place them in separate directory so that noone runs them unintentionally when running the unitt ests.

Additionally, several test cases are exceptionally long. Those are run iff the environmental variable `SLOW_BENCHMARKS` is set (to any value). So, to run "fast" benchmarks:

```bash
pytest benchmarks/
```

and to run all benchmarks, including the slow ones:

```bash
SLOW_BENCHMARKS=1 pytest benchmarks/
```

The tests are implemented using the `pytest-benchmark` library, with the default settings (the number of iterations and repetitions are determined by `pytest-benchmark`).

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [x] I have updated documentation.

### CODE REVIEW

File: benchmarks/test_get_qsp_time_evolution_program.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+from benchq.problem_ingestion import (
+    generate_jw_qubit_hamiltonian_from_mol_data,
+    get_vlasov_hamiltonian,
+)
+from benchq.problem_ingestion.hamiltonian_generation import fast_load_qubit_op
+from benchq.problem_ingestion.molecule_instance_generation import (
+    generate_hydrogen_chain_instance,
+)
+
+SKIP_SLOW = pytest.mark.skipif(
```
Comment (AthenaCaesura) [inline benchmarks/test_get_qsp_time_evolution_program.py:16]: Good solution here!

**Review by AthenaCaesura — APPROVED** [review]

**Comment by AthenaCaesura** [comment]
I've approved this PR because it would work by itself. But can I ask you to add the tests that I have written up for the other parts of benchq?

**Review by mstechly — CHANGES_REQUESTED** [review]
Please add information how to use those tests to Readme.

**Comment by dexter2206** [comment]
> I've approved this PR because it would work by itself. But can I ask you to add the tests that I have written up for the other parts of benchq?

Don in [3dd8a15](https://github.com/zapatacomputing/benchq/pull/38/commits/3dd8a15e9b4f15117a31466928f6661add88697e)

**Comment by dexter2206** [comment]
> 533e6d4

README updated in [533e6d4](https://github.com/zapatacomputing/benchq/pull/38/commits/533e6d4d976c3a10c19ddbd5e8ed27e08ed656bf)

File: src/benchq/problem_ingestion/hamiltonian_generation.py [hunk 2]
```diff
@@ -92,11 +93,8 @@ def generate_1d_heisenberg_hamiltonian(N):
 
 
 def fast_load_qubit_op(file):
-    if isinstance(file, str):
-        with open(file, "r") as f:
-            data = json.load(f)
-    else:
-        data = json.load(file)
+    with ensure_open(file, "r") as f:
+        data = json.load(f)
```
[the remark this answers is not in the document]
Comment (dexter2206) [reply src/benchq/problem_ingestion/hamiltonian_generation.py:97]: I'd like to keep this PR only about performance tests and not performance improvements. However, I'm certainly not opposed to introducing rapidjson if we find out that reading JSON file is one of the bottlenecks.

**Review by mstechly — APPROVED** [review]

**Pull request merged**